src/operations/add_user.ts

```diff
@@ -56,25 +61,22 @@ export class AddUserOperation extends CommandOperation<Document> {
       );
     }
 
-    // Get additional values
-    let roles: string[] = [];
-    if (Array.isArray(options.roles)) roles = options.roles;
-    if (typeof options.roles === 'string') roles = [options.roles];
-
-    // If not roles defined print deprecated message
-    // TODO: handle deprecation properly
-    if (roles.length === 0) {
-      console.warn('Creating a user without roles is deprecated in MongoDB >= 2.6');
+    let roles;
+    if (!options.roles || (Array.isArray(options.roles) && options.roles.length === 0)) {
+      emitWarningOnce(
+        'Creating a user without roles is deprecated. Defaults to "root" if db is "admin" or "dbOwner" otherwise'
+      );
+      if (db.databaseName.toLowerCase() === 'admin') {
+        roles = ['root'];
+      } else {
+        roles = ['dbOwner'];
+      }
+    } else {
+      roles = Array.isArray(options.roles) ? options.roles : [options.roles];
     }
 
-    // Check the db name and add roles if needed
-    if (
-      (db.databaseName.toLowerCase() === 'admin' || options.dbName === 'admin') &&
-      !Array.isArray(options.roles)
-    ) {
-      roles = ['root'];
-    } else if (!Array.isArray(options.roles)) {
-      roles = ['dbOwner'];
+    if (roles.length === 0) {
+      // If not roles defined throw error
```

Comment:
```suggestion
      // If no roles defined throw error
```

Reply:
Actually this if statement does nothing with the added logic above it, so whoosh, it is gone! 🎉